### ISSUE TITLE
[BUG] if in confg file -multiservice is missing do not notify if append request has Service-Fiware diferent at default value

### ISSUE THREAD

**Issue by iariasleon** [telefonicaid/fiware-orion#1898]
Following an incidence of StackOverflow: http://stackoverflow.com/questions/35683758/orion-contextbroker-dont-notifies-correctly-when-data-comes-from-iotagent

I have realized that if in the configuration file does not define the `-multiservice` parameter and having a Subcription in orion (default db), if we send an append request (with value changed) without `Fiware-Service` header a notification is sent, but if we send an append request  (with value changed) and any `Fiware-Service` header (default db include), the notification is not sent.

#### CB config
```
contextBroker -port 1026 -logDir /var/log/contextBroker -pidpath /var/run/contextBroker/contextBroker.pid -dbhost localhost -db orion -t 0-255 -logLevel DEBUG
```

#### Subcription
```
http://localhost:1026/NGSI10/subscribeContext
Accept: application/json
Content-Type: application/json
{
    "entities": [
        {
            "type": "",
            "isPattern": "false",
            "id": "D000"
        }
    ],
    "attributes": [
        "Temperature0", "Humidity0"
    ],
    "reference": "http://localhost:8090/notify",
    "duration": "P1M",
    "notifyConditions": [
        {
            "type": "ONCHANGE",
            "condValues": [
                "Temperature0", "Humidity0"
            ]
        }
    ],
  "throttling": "PT1S"
  
}
```

#### First Append request
this request is notified
```
http://localhost:1026/NGSI10/updateContext	
Content-Type: application/json
{
"contextElements":[{
    "type":"thing",
    "isPattern":"false",
    "id":"D000",
    "attributes":[
         {
              "name":"Temperature0",
              "type":"Number",
              "value":"4555"
        }
     ]
  }],
"updateAction":"APPEND"
}
```

#### Second Append request
this request is **NOT** notified
```
http://localhost:1026/NGSI10/updateContext	
Content-Type: application/json
Fiware-Service: 
{  
   "updateAction":"APPEND",
   "contextElements":[  
      {  
         "id":"D000",
         "type":"thing",
         "isPattern":"false",
         "attributes":[  
            {  
               "name":"Temperature0",
               "type":"Number",
               "value":"456"
            }
         ]
      }
   ]
}
```

#### Third Append request
this request is **NOT** notified
```
http://localhost:1026/NGSI10/updateContext	
Content-Type: application/json
Fiware-Service: fiware
{  
   "updateAction":"APPEND",
   "contextElements":[  
      {  
         "id":"D000",
         "type":"thing",
         "isPattern":"false",
         "attributes":[  
            {  
               "name":"Temperature0",
               "type":"Number",
               "value":"45667"
            }
         ]
      }
   ]
}
```

**Comment by iariasleon** [comment]
Tested in:
```
  "version" : "0.28.0-next",
  "git_hash" : "1a5983818765ab91959f31df190d3e575292bda7",
```

**Comment by iariasleon** [comment]
Also tested with `"updateAction": "UPDATE"` in all append requests (remember create an entity previously) and the behavior is the same.

**Comment by iariasleon** [comment]
Always the entity is updated in `orion` (default), but in the cases with Service-Fiware header (second and third) looking for a subscription with service

**Comment by iariasleon** [comment]
With `-noCache` parameter activate, notity all requests, without and with `Service_fiware` header.
```
contextBroker -port 1026 -logDir /var/log/contextBroker -pidpath /var/run/contextBroker/contextBroker.pid -dbhost localhost -db orion -noCache -t 0-255 -logLevel DEBUG
```

**Comment by fgalan** [comment]
The following answer at SOF needs to be edited after this issue get solved: http://stackoverflow.com/questions/35683758/orion-contextbroker-dont-notifies-correctly-when-data-comes-from-iotagent/35944217#35944217

**Comment by fgalan** [comment]
Fixed in PR #2029. @iariasleon please have a look in order to close it if it is ok.

**Comment by iariasleon** [comment]
LGTM tested in:
```
  "version" : "1.0.0-next",
  "git_hash" : "75e7e8f31cdd5aa88e8516b90a41ba1127b6e0b4"
```
 CB Started without `-multiservice` and without/with `-noCache` cli parameters and all notifications are received
